tests/suites/storage/model_storage_filesystem.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# Test that model config default storage type is used when not specified.
+# This test verifies that when storage type is not explicitly specified,
+# Juju uses the default storage filesystem type from model configuration.
+run_model_storage_filesystem() {
+	echo
+
+	model_name="model-storage-filesystem"
+	file="${TEST_DIR}/test-${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+
+	# Run filesystem test for all providers.
+	test_default_filesystem_storage "${model_name}"
+
+	destroy_model "${model_name}"
+}
+
+# Test filesystem storage uses the model's default storage type.
+test_default_filesystem_storage() {
+	local model_name=$1
+
+	echo "Testing model config filesystem storage type with tmpfs"
+
+	# Set default filesystem storage type in model config.
+	juju model-config -m "${model_name}" storage-default-filesystem-source=tmpfs
```

Comment:
1G might stretch the allowed memory for the vm running the test
maybe make the size smaller, eg 100M. it doesn't need to be large

Reply:
Fixed.